src/Stratis.Bitcoin/Consensus/ConsensusExceptions.cs

```diff
@@ -84,4 +84,15 @@ public IntegrityValidationFailedException(IPEndPoint peer, ConsensusError error,
             this.BanDurationSeconds = banDurationSeconds;
         }
     }
+
+    public class BlockNotFoundException : ConsensusException
```

Comment:
I think we should move this Exception to `Stratis.Bitcoin.Features.RPC.Exceptions`? ConsensusExceptions are exceptions thrown by the consensus mechanism when downloading and validating new blocks from other peers.

Reply:
The issue is that `ConsensusController` is in `Stratis.Bitcoin.Features.Consensus`, I can't reference exception from another feature here unless we move exception into core library.

Reply:
Looks wrong to me to use `ConsensusException` in RPC, where does this get thrown?